codec-http2/src/main/java/io/netty/handler/codec/http2/Http2CodecUtil.java

```diff
@@ -190,6 +192,148 @@ public static void writeFrameHeader(ByteBuf out, int payloadLength, byte type,
     }
 
     /**
+     * Provides the ability to associate the outcome of multiple {@link ChannelPromise}
+     * objects into a single {@link ChannelPromise} object. This can be done with relatively low overhead
+     * if the number of {@link ChannelPromise} objects is known a priori.
+     */
+    static class SimpleChannelPromiseAggregator implements ChannelFutureListener {
+        private final ChannelHandlerContext ctx;
+        private final ChannelPromise promise;
+        private final int expectedCount;
+        private int completedCount;
+
+        SimpleChannelPromiseAggregator(ChannelHandlerContext ctx, int expectedCount, ChannelPromise promise) {
```

Comment:
This is an option, but then all objects have to be created up front before any are used. It will change the structure of the code a bit (user will have to know that they have to use `newPromise()` `x` times before they can safety use any of the promises.